pkg/api/helper/helpers.go

```diff
@@ -113,9 +113,18 @@ func IsStandardContainerResourceName(str string) bool {
 }
 
 // IsOpaqueIntResourceName returns true if the resource name has the opaque
-// integer resource prefix.
+// integer resource prefix, or it is another unknown resource name not in the
+// default namespace.
 func IsOpaqueIntResourceName(name api.ResourceName) bool {
-	return strings.HasPrefix(string(name), api.ResourceOpaqueIntPrefix)
+	return IsCustomNamespacedResourceName(name)
```

Comment:
@jiayingz good catch; the predicate as written would match `alpha.kubernetes.io/` resources.

Another option could be to match the substring instead of the prefix.

```go
func IsCustomNamespacedResourceName(name api.ResourceName) bool {
	return strings.Contains(string(name), "/") &&
		!strings.Contains(string(name), api.ResourceDefaultNamespacePrefix)
}
```